extensions/markdown-language-features/src/features/workspaceSymbolProvider.ts

```diff
@@ -56,16 +84,12 @@ class VSCodeWorkspaceMarkdownDocumentProvider extends Disposable implements Work
 
 		this._watcher.onDidChange(async resource => {
 			const document = await this.getMarkdownDocument(resource);
-			if (document) {
```

Comment:
Can you please make these changes in a separate PR? I'd like to keep this one scoped to just batching the requests